modules/v1/workspaces/controllers/ArticleController.php

```diff
@@ -0,0 +1,179 @@
+            }
+            $workspaceId = $parentFolder->workspace_id;
+        }
+
+        $article = new ArticleResource();
+
+        $article->title = $request->post('title');
+        $article->body = $request->post('body');
+        $article->workspace_id = $workspaceId;
+        $article->is_folder = $isArticle ? 0 : 1;
+
+        if ((!$article->load($request->post(), '')) && !$article->validate()) {
+            return $this->validationError($article->getFirstErrors());
+        }
+
+        if (!$articleId) {
+            if (!$article->makeRoot()) {
+                $this->validationError($article->getFirstErrors());
+            }
+        } else {
+            $article->parent_id = $articleId;
+            if (!$article->appendTo($parentFolder)) {
+                $this->validationError($article->getFirstErrors());
+            }
+        }
+
+        return $this->response($article, 201);
+    }
+
+    /**
+     * Get breadcrumb for article view page
+     *
+     * @param $articleId
+     * @return array
+     * @throws ValidationException
+     */
+    public function actionGetBreadCrumb($articleId)
+    {
+        $article = ArticleResource::find()->byId($articleId)->one();
+
+        if (!$article) {
+            throw new ValidationException(Yii::t('app', 'This article not exist'));
+        }
+
+        $workspace = $article->workspace;
+        $breadCrumb[] = [
+            'text' => Yii::t('app', 'My Workspaces'),
+            'to' => [
+                'name' => 'workspace',
+            ]
+        ];
+
+        $breadCrumb[] = [
+            'text' => $workspace->abbreviation ?: $workspace->name,
+            'to' => [
+                'name' => 'workspace.view',
+                'params' => [
+                    'id' => $workspace->id
+                ]
+            ]
+        ];
+
+        $parents = $article->parents()->all();
+
+        foreach ($parents as $parent) {
+            $breadCrumb[] = [
+                'text' => $parent->title,
+                'to' => [
+                    'name' => 'article.view',
+                    'params' => [
+                        'id' => $parent->id
+                    ]
+                ]
+            ];
+        }
+        $breadCrumb[] = [
+            'text' => $article->title
+        ];
+
+        return $breadCrumb;
+    }
+
+    /**
+     * Get articles by parent
+     *
+     * @param $articleId
+     * @return ActiveDataProvider
+     */
+    public function actionByParent($articleId)
+    {
+        $query = ArticleResource::find()->byParentId($articleId);
+
+        return new ActiveDataProvider([
+            'query' => $query
+        ]);
+    }
+
+    /**
+     * Add articles to favourites
+     *
+     * @return mixed
+     * @throws ValidationException
+     */
+    public function actionAddToFavourites()
+    {
+        $userModel = Yii::$app->user->identity;
+
+        $userModel->favourites = json_encode(Yii::$app->request->post(), true);
```

Comment:
@skavt It is better to use `yii\helpers\Json::encode` function instead of php core function.